File: AuthPreference/authPrefLibrary/src/main/java/com/sachith/authpreflibrary/AuthPreference.java
package com.sachith.authpreflibrary;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.ArrayList;

public class AuthPreference {

    public static void authUserdata(Context context, int userId, boolean state){
        SharedPreferences sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("userId", userId);
        editor.putBoolean("status", state);
        editor.apply();
    }

    public static void saveUserHeaders(Context context, ArrayList<Headers> headers){
        SharedPreferences sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        for(int i = 0; i < headers.size(); i++){
            editor.putString(headers.get(i).getHeader(), headers.get(i).getValue());
        }
        editor.apply();
    }

    public static Boolean getUserState(Context context){
        SharedPreferences sharedPreferences = context.getSharedPreferences("User",Context.MODE_PRIVATE);
        return sharedPreferences.getBoolean("status", false);
    }

    public static int getUserId(Context context){
        SharedPreferences sharedPreferences = context.getSharedPreferences("User",Context.MODE_PRIVATE);
        return sharedPreferences.getInt("userId", 0);
    }

    public static ArrayList<Headers> getUserHeaders(Context context, ArrayList<String> headerNames){
        SharedPreferences sharedPreferences = context.getSharedPreferences("User",Context.MODE_PRIVATE);
        ArrayList<Headers> headers = new ArrayList<>();
        for(int j =0 ; j<headerNames.size(); j++){
            headers.add(new Headers(headerNames.get(j), sharedPreferences.getString(headerNames.get(j), "")));
        }
        return headers;
    }

}

File: auth-preference/src/main/java/io/github/sachithariyathilaka/AuthPreference.kt
package io.github.sachithariyathilaka

import android.content.Context
import io.github.sachithariyathilaka.resource.Headers
import io.github.sachithariyathilaka.resource.UserDetail

/**
 * Main class for the library.
 *
 * @author  Sachith Ariyathilaka
 * @version 1.0.0
 * @since   2024/03/24
 */
object AuthPreference {

    /**
     * Static method for save user details.
     *
     * @param   context the context
     * @param   userDetail the user details
     */
    fun saveUserDetails(context: Context, userDetail: UserDetail) {
        val sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE)

        val editor = sharedPreferences.edit()
        editor.putString("userId", userDetail.userId)
        editor.putBoolean("status", userDetail.status)

        editor.apply()
    }

    /**
     * Static method for save user headers.
     *
     * @param   context the context
     * @param   headers the header list
     */
    fun saveHeaders(context: Context, headers: List<Headers>) {
        val sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE)

        val editor = sharedPreferences.edit()
        for (i in headers.indices)
            editor.putString(headers[i].header, headers[i].value)

        editor.apply()
    }

    /**
     * Static method for get user details.
     *
     * @param   context the context
     *
     * @return  the user details
     */
    fun getUserDetails(context: Context): UserDetail? {
        val sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE)
        return sharedPreferences.getString("userId", "")?.let { UserDetail(it, sharedPreferences.getBoolean("status", false)) }
    }

    /**
     * Static method for get user headers.
     *
     * @param   context the context
     * @param   headerNames the header names
     *
     * @return  the headers
     */
    fun getHeaders(context: Context, headerNames: List<String?>): ArrayList<Headers> {
        val sharedPreferences = context.getSharedPreferences("User", Context.MODE_PRIVATE)
        val headers = ArrayList<Headers>()

        for (j in headerNames.indices) headers.add(
            Headers(
                headerNames[j]!!, sharedPreferences.getString(
                    headerNames[j], ""
                )!!
            )
        )

        return headers
    }
}
